### PR TITLE
Image tools tests redux

### PR DESCRIPTION
**Pull request by wjwwood**
replacement for https://github.com/ros2/demos/pull/48 that does not use a movie file, but generates burgers instead (thanks @codebot).

screen shot 2016-05-24 at 8 25 42 pm

### CODE REVIEW

**Comment by wjwwood** [comment]
CI:

- Linux [Build Status](http://ci.ros2.org/job/ci_linux/1342/)
- OS X [Build Status](http://ci.ros2.org/job/ci_osx/1051/)
- Windows [Build Status](http://ci.ros2.org/job/ci_windows/1355/)

**Comment by wjwwood** [comment]
Btw, this new test is disabled for connext, connext dynamic, and fastrtps for various known issues. I put a comment in the cmake which disables it.

**Comment by wjwwood** [comment]
Ok, it's broken on Windows and OS X. OS X is a problem with how OpenCV is linked in Homebrew and I don't know how to fix it generically. Windows is a mystery too. It's got something to do with how I'm using `file(GENERATE ...)`. I'm going to put this on hold until after the alpha. I've wasted way too much time on it. If someone else wants to pick it up before then, that'd be great.

**Comment by wjwwood** [comment]
CI:

- Linux:
  - [Build Status](http://ci.ros2.org/job/ci_linux/1716/)
- OS X:
  - [Build Status](http://ci.ros2.org/job/ci_osx/1282/)
- Windows:
  - [Build Status](http://ci.ros2.org/job/ci_windows/1638/)

File: image_tools/CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +53,56 @@ call_for_each_rmw_implementation(targets GENERATE_DEFAULT)
...
+      "${rmw_implementation} " STREQUAL "rmw_connext_dynamic_cpp " OR
+      # TODO(wjwwood): fastrtps fails with:
+      #
+      #   uncaught exception of type std::runtime_error: vector overcomes the maximum length
+      #
+      # We can remove this disable for fastrtps when that's fixed.
+      "${rmw_implementation} " STREQUAL "rmw_fastrtps_cpp "
+    )
+      message(STATUS "Skipping tests for '${rmw_implementation}'")
+      return()
```
Comment (dirk-thomas) [inline image_tools/CMakeLists.txt]: I assume the purpose of this return is to skip the current rmw implementation. But `return` doesn't work like that in a `macro`. Please see https://cmake.org/cmake/help/v3.5/command/return.html

**Comment by dirk-thomas** [comment]
It also looks like this change broke the Windows build when using OpenSplice: see http://ci.ros2.org/job/ci_windows/1650/console

**Comment by wjwwood** [comment]
I'll look at it, but the CI jobs don't test OpenSplice any longer.

File: image_tools/CMakeLists.txt (hunk 1 above)
Comment (wjwwood) [inline image_tools/CMakeLists.txt]: This should be resolved in https://github.com/ros2/demos/pull/78.

**Comment by dirk-thomas** [comment]
I created a ticket for it: #79

**Pull request merged**